src/gtest/test_sidechain_certificate_quality.cpp

```diff
@@ -323,7 +323,44 @@ TEST_F(SidechainsMultipleCertsTestSuite, V2Cert_HigherQuality_SameEpoch_Sidechai
     EXPECT_TRUE(sidechain.balance                     == initialScState.balance);
 }
 
-TEST_F(SidechainsMultipleCertsTestSuite, V2Cert_LowerQuality_SameEpoch_SidechainIsNotUpdated) {
+TEST_F(SidechainsMultipleCertsTestSuite, V2Cert_SameEpoch_LowerQuality_SidechainIsNotUpdated) {
+    uint256 scId = uint256S("aaa");
+
+    // setup sidechain initial state...
+    CSidechain initialScState;
+    initialScState.fixedParams.withdrawalEpochLength = 0;
+    initialScState.fixedParams.version = 2;
+    initialScState.creationBlockHeight = 400;
+    initialScState.lastTopQualityCertHash = uint256S("cccc");
+    initialScState.lastTopQualityCertQuality = 1;
+    initialScState.lastTopQualityCertReferencedEpoch = 1987;
+    initialScState.lastTopQualityCertBwtAmount = 50;
+    initialScState.balance = CAmount(100);
+    initialScState.InitScFees();
+    storeSidechainWithCurrentHeight(*sidechainsView, scId, initialScState, initialScState.creationBlockHeight);
+
+    //Insert high quality Certificate
+    CMutableScCertificate highQualityCert;
+    highQualityCert.scId        = scId;
+    highQualityCert.epochNumber = initialScState.lastTopQualityCertReferencedEpoch;
+    highQualityCert.quality     = 0;
+    highQualityCert.addBwt(CTxOut(CAmount(90), dummyScriptPubKey));
+    CAmount highQualityCert_TotalBwtAmount = CScCertificate(highQualityCert).GetValueOfBackwardTransfers();
+
+    //test
+    //sidechainsView->UpdateSidechain(highQualityCert, dummyUndo);
+    ASSERT_FALSE(sidechainsView->UpdateSidechain(highQualityCert, dummyUndo, sidechainsView->GetHeight()+1));
```

Comment:
I guess `highQualityCert` should be named `lowQualityCert`, am I correct?

Reply:
Updated in dfeb52fab30e3b941b44631be6cdf8a95f317a67